### PR TITLE
ADD docs for composition feature

### PR DESCRIPTION
**Pull request by ndelangen**
## What I did

#10528 #9210 #10504 

I added docs

(some of the described features aren't fully implemented yet)

- discuss mapper & add tests
- add auto-loading feature

### CODE REVIEW

File: docs/src/pages/configurations/composition/index.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+When you compose a storybook into your own, you get **all** the stories, in their original format. You may not agree with some names, or you might want to filter or sort some stories to your liking.
+
+You can add a 'mapper' to storybook that will be used to transform the received stories to something of your liking.
+
+```js
+import { addons } from '@storybook/addons';
+
+addons.setConfig({
+  mapper: (ref, story) => {
```
Comment (shilman) [inline docs/src/pages/configurations/composition/index.md:100]: Can we rename this to `storyMapper`?

File: docs/src/pages/configurations/composition/index.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+- you have a need for multiple frameworks
+- you have multiple teams using separate storybooks
+- you are using community component libraries
+
+---
+
+## Composing a storybook
+
+To compose another storybook into your, all you need is the URL of storybook.
```
Comment (shilman) [inline docs/src/pages/configurations/composition/index.md:16]:
```suggestion
To compose another storybook into yours, all you need is the URL of storybook.
```

**Review by shilman — APPROVED** [review]
LGTM! A few suggestions inline

File: docs/src/pages/configurations/composition/index.md (hunk 1 above)
Comment (ndelangen) [reply docs/src/pages/configurations/composition/index.md:100]: I'll work on this today

Comment (ndelangen) [reply docs/src/pages/configurations/composition/index.md]: done!

**Comment by ndelangen** [comment]
Waiting for https://github.com/storybookjs/storybook/pull/10780 to be approved and merged before merging this.

**Pull request merged**